### PR TITLE
Refactor resource snapshot and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by Diptipowervs** [comment]
--- PASS: TestAccIBMPIInstanceSnapshotbasic (1859.75s)
PASS

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 1]
```diff
@@ -35,54 +34,54 @@ func ResourceIBMPISnapshot() *schema.Resource {
...
 				Description: "Unique name of the snapshot",
 			},
-			helpers.PIInstanceName: {
+			Arg_InstanceName: {
 				Type:        schema.TypeString,
 				Required:    true,
 				Description: "Instance name / id of the pvm",
 			},
-			helpers.PIInstanceVolumeIds: {
+			Arg_InstanceVolumeIds: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:47]: Should be`PIInstanceVolumeIds                 = "pi_volume_ids"`
but changed to`Arg_InstanceVolumeIds                   = "pi_instance_volume_ids"`
Change value to be `pi_volume_ids`

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 2]
```diff
@@ -96,17 +95,17 @@ func resourceIBMPISnapshotCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.FromErr(err)
 	}
 
-	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
-	instanceid := d.Get(helpers.PIInstanceName).(string)
-	volids := flex.ExpandStringList((d.Get(helpers.PIInstanceVolumeIds).(*schema.Set)).List())
-	name := d.Get(helpers.PISnapshotName).(string)
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	instanceid := d.Get(Arg_InstanceName).(string)
+	volids := flex.ExpandStringList((d.Get(Arg_InstanceVolumeIds).(*schema.Set)).List())
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:100]: Need to be updated

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 3]
```diff
@@ -96,17 +95,17 @@ func resourceIBMPISnapshotCreate(ctx context.Context, d *schema.ResourceData, me
...
+	name := d.Get(Arg_SnapshotName).(string)
 
 	var description string
-	if v, ok := d.GetOk("pi_description"); ok {
+	if v, ok := d.GetOk(Attr_Description); ok {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:104]: Should be `Arg_Description `

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 4]
```diff
@@ -145,18 +144,18 @@ func resourceIBMPISnapshotRead(ctx context.Context, d *schema.ResourceData, meta
...
 	}
 
-	d.Set(helpers.PISnapshotName, snapshotdata.Name)
-	d.Set("snapshot_id", *snapshotdata.SnapshotID)
-	d.Set("status", snapshotdata.Status)
-	d.Set("creation_date", snapshotdata.CreationDate.String())
-	d.Set("volume_snapshots", snapshotdata.VolumeSnapshots)
-	d.Set("last_update_date", snapshotdata.LastUpdateDate.String())
+	d.Set(Arg_SnapshotName, snapshotdata.Name)
+	d.Set(Arg_SnapshotID, *snapshotdata.SnapshotID)
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:154]: Should be an attribute

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 5]
```diff
@@ -174,10 +173,10 @@ func resourceIBMPISnapshotUpdate(ctx context.Context, d *schema.ResourceData, me
 		return diag.FromErr(err)
 	}
 
-	client := st.NewIBMPISnapshotClient(ctx, sess, cloudInstanceID)
+	client := instance.NewIBMPISnapshotClient(ctx, sess, cloudInstanceID)
 
-	if d.HasChange(helpers.PISnapshotName) || d.HasChange("description") {
-		name := d.Get(helpers.PISnapshotName).(string)
+	if d.HasChange(Arg_SnapshotName) || d.HasChange("description") {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:178]: Use const for description and it should `pi_description`

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 6]
```diff
@@ -174,10 +173,10 @@ func resourceIBMPISnapshotUpdate(ctx context.Context, d *schema.ResourceData, me
...
+		name := d.Get(Arg_SnapshotName).(string)
 		description := d.Get("description").(string)
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:180]: same as above

**Review by michaelkad — CHANGES_REQUESTED** [review]

File: ibm/service/power/resource_ibm_pi_snapshot.go (hunk 1 above)
Comment (Diptipowervs) [reply ibm/service/power/resource_ibm_pi_snapshot.go]: Changes made

File: ibm/service/power/resource_ibm_pi_snapshot.go (hunk 2 above)
Comment (Diptipowervs) [reply ibm/service/power/resource_ibm_pi_snapshot.go]: done

**Comment by Diptipowervs** [comment]
close PR

**Pull request closed without merging**